subsys/net/lib/nrf_cloud/src/nrf_cloud_agps.c

```diff
@@ -680,6 +651,7 @@ int nrf_cloud_agps_process(const char *buf, size_t buf_len, const int *socket)
 	size_t parsed_len = 0;
 	uint8_t version;
 
+	ARG_UNUSED(socket);
```

Comment:
if a socket is no longer useable for the new interface, let's just remove it from `nrf_cloud_agps_process`?

Reply:
Yes I'm very aware of this. The reason to do like this is avoid the enormous changes in other app/lib that calls this API. In fact there are more change than this. For API nrf_cloud_agps_request(), we should NOT use the data structure defined in gps.h any more, but use the one defined in new GNSS API header. I'd like to leave all those changes be done by a new PR and by someone not myself.

Reply:
As I mentioned in my other comment, removing support for the GNSS socket and GPS driver needs further consideration, i.e. when and how it's done.